### PR TITLE
Support OpenBSD on arm64

### PR DESCRIPTION
**Pull request by knightjoel**
### Issues:
n/a

### Description of changes: 
This PR basically provides `OPENSSL_cpuid_setup()` for OpenBSD/aarch64.

### Call-outs:
The cpu_aarch64_openbsd.c file is lifted from OpenBSD's patch set to Chromium.

### Testing:
A standalone build of aws-lc now succeeds as does building aws-lc as part of the Python CRT bindings. Tested on OpenBSD 7.3.

By submitting this pull request, I confirm that my contribution is made under the terms of the ISC license.

### CODE REVIEW

File: crypto/fipsmodule/cpucap/cpu_aarch64_openbsd.c [hunk 1]
```diff
@@ -0,0 +1,57 @@
+/* Copyright (c) 2022, Robert Nagy <robert@openbsd.org>
```
Comment (dkostic) [inline crypto/fipsmodule/cpucap/cpu_aarch64_openbsd.c:1]: is this file copied from OpenBSD?

File: crypto/fipsmodule/cpucap/cpu_aarch64_openbsd.c [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+#include <stdio.h>
+
+#include <openssl/arm_arch.h>
+
+#include "internal.h"
+
+extern uint32_t OPENSSL_armcap_P;
+
+void OPENSSL_cpuid_setup(void) {
+  int isar0_mib[] = { CTL_MACHDEP, CPU_ID_AA64ISAR0 };
```
Comment (dkostic) [inline crypto/fipsmodule/cpucap/cpu_aarch64_openbsd.c:32]: can we please add a reference where all these macros come from?

Comment (knightjoel) [reply crypto/fipsmodule/cpucap/cpu_aarch64_openbsd.c:32]: I can, but I'm not sure the extent of what you're asking. Is it just for the macros on line 32? I don't see references in any of the other cpu_*.c files for macros they use.

File: crypto/fipsmodule/cpucap/cpu_aarch64_openbsd.c (hunk 1 above)
Comment (knightjoel) [reply crypto/fipsmodule/cpucap/cpu_aarch64_openbsd.c:1]: This file is copied from the set of patches which OpenBSD applies to Chromium in order to build Chromium on OpenBSD. The author of this particular patch happens to be an OpenBSD committer.

File: crypto/fipsmodule/cpucap/cpu_aarch64_openbsd.c (hunk 2 above)
Comment (dkostic) [reply crypto/fipsmodule/cpucap/cpu_aarch64_openbsd.c:32]: just a comment that they are defined in "X" header file is sufficient.

Comment (knightjoel) [reply crypto/fipsmodule/cpucap/cpu_aarch64_openbsd.c:35]: Does it look like what you're expecting now?

File: crypto/fipsmodule/cpucap/cpu_aarch64_openbsd.c [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+#include <openssl/arm_arch.h>
+
+#include "internal.h"
+
+extern uint32_t OPENSSL_armcap_P;
+
+void OPENSSL_cpuid_setup(void) {
+  /* CTL_MACHDEP from sys/sysctl.h
+   * CPU_ID_AA64ISAR0 from machine/cpu.h
+   */
```
Comment (dkostic) [inline crypto/fipsmodule/cpucap/cpu_aarch64_openbsd.c:34]:
```suggestion
  // CTL_MACHDEP from sys/sysctl.h
  // CPU_ID_AA64ISAR0 from machine/cpu.h
```

File: crypto/fipsmodule/cpucap/cpu_aarch64_openbsd.c (hunk 2 above)
Comment (dkostic) [reply crypto/fipsmodule/cpucap/cpu_aarch64_openbsd.c:34]: looks good. I just changed the comment style and updated the PR with the main branch.

**Review by dkostic — APPROVED** [review]

**Review by skmcgrail — APPROVED** [review]

**Pull request merged**